build.sbt

```diff
@@ -507,7 +512,13 @@ lazy val millCodegenPlugin = projectMatrix
     name := "mill-codegen-plugin",
     crossVersion := CrossVersion
       .binaryWith(s"mill${millPlatform(Dependencies.Mill.millVersion)}_", ""),
-    libraryDependencies ++= Seq(
+    libraryDependencies := codegenDeps.value ++ Seq(
```

Comment:
I find it weird that you'd have to specify `codegenDeps.value` here considering this module depends on the codegen module, and therefore should benefit from its dependencies without having to add them here.

Reply:
My understanding, is that the sbt `:=` operator _discards_ and overwrites any inherited value to this point. 

Which begs the question : why not stay with the `++=` operator for the "non-provided" scope.  The answer to that, was that the "logic" then became hard for me to follow (weak sbt again, perhaps). My choice was then to be as explicit as possible that something quite deliberate and at least a little non-standard was going on, by using the `:=` operator in both places. 

It may not be the right approach, but it set out (for me) the logic as clearly as possible...  I'm not a confident sbt user, so certainly willing to stand corrected.

Reply:
No problem, I'll try and find the time to try it locally before making a decision of whether to approve as-is or not. Regardless, thank you very much for the contribution !

Reply:
Yes, I have to say that I think this is a surprisingly risky, potentially high maintenance sort of change. In the interim, it appears that Tobias at mill has done something which might sort this out millside - maybe it would be preferable to see if the below mill PR sorts it out in preference to merging this?

https://github.com/com-lihaoyi/mill/pull/3189

As ever, thank you for taking the time to review 🙏